CHANGES

```diff
@@ -1,8 +1,15 @@
 ===== dev =====
 
+====== Breaking ======
```

Comment:
I am neutral on whether deprecation counts as breaking changes and can work either way. :grin: I see that it could be counted as breaking as we want the users to modify their code, but I also see that it is not breaking because they can choose not to. Happy to update the PR once a consensus is reached.

Reply:
I guess it breaks the code that is compiled with warning-as-failure.

IIRC, `dune`'s release mode sets this option to false and dune-built opam packages are supposed to be built in release mode. So we shouldn't break too many packages. But I think that deprecation of such a central function will probably break some development versions of some projects.

I don't know to what extent this counts as breaking: it depends on a compilation option that is set differently by different toolchains on different occasions.

Reply:
> IIRC, `dune`'s release mode sets this option to false and dune-built opam packages are supposed to be built in release mode. So we shouldn't break too many packages.

It's not breaking because of this. Doing this change does not affect the "main" opam-based builds, and anything similar to them. In addition, IIRC, if one uses the various vendoring stanzas of Dune, vendored subprojects are also built with warnings as warnings.

We used to routinely introduce deprecations in Lwt without a major version increment, usually three months or so before such an increment, which would then commit the actual breaking change warned in the deprecation messages.

Reply:
It seems consensus was reached.